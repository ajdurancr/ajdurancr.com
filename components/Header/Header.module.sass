@import '../../styles/_variables'

$headerMarginMobile: 3.2rem
$headerMarginTablet: 6.4rem

.header
  display: flex
  flex-direction: column
  height: $menuHeightMobile
  left: 0 
  position: fixed
  top: 0
  width: 100%
  z-index: 1

.container
  margin: 0 $headerMarginMobile
  width: calc(100% - #{$headerMarginMobile}*2)

  @media only screen and (min-width: $size-tablet)
    width: calc(100% - #{$headerMarginTablet}*2)
    margin: 0 $headerMarginTablet

.topMenu
  display: flex
  justify-content: space-between
  min-height: 5rem
  align-items: center
  padding: 1.6rem 0
  width: 100%

  .logo
    display: flex
    max-width: 15rem

    @media only screen and (min-width: $size-tablet)
      max-width: 20rem

