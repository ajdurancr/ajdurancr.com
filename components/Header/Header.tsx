import React from 'react'
import classnames from 'classnames'

import type { CfComponent, CfMenu } from '../../types/contentful'
import { Logo } from '../Logo'
import { Menu } from '../Menu'
import { Component } from '../Component'
import styles from './Header.module.sass'

type HeaderProps = {
  menu: CfMenu
  pageUrl: string
  socialMedia?: CfComponent
}

const Header: React.FC<HeaderProps> = ({ pageUrl, menu, socialMedia }) => (
  <div className={classnames(styles.header, 'header')}>
    <div className={styles.container}>
      <div className={styles.topMenu}>
        <Logo className={styles.logo} />
        {socialMedia && <Component component={socialMedia} />}
      </div>
      <Menu pageUrl={pageUrl} menu={menu} />
    </div>
  </div>
)

export { Header }
