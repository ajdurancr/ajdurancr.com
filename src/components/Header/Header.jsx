import React from 'react'
import propTypes from 'prop-types'
import classnames from 'classnames'
import get from 'lodash.get'

import { Logo, Menu, Component } from '..'
import styles from './Header.module.sass'

const Header = ({ pageUrl, menu, socialMedia }) => (
  <div className={classnames(styles.header, 'header')}>
    <div className={styles.container}>
      <div className={styles.topMenu}>
        <Logo className={styles.logo} />
        <Component {...(socialMedia)} />
      </div>
      <Menu pageUrl={pageUrl} menu={get(menu, 'fields')} />
    </div>
  </div>
)

Header.defaultProps = {
  socialMedia: {},
}

Header.propTypes = {
  menu: propTypes.object.isRequired,
  pageUrl: propTypes.string.isRequired,
  socialMedia: propTypes.object,
}

export default Header
