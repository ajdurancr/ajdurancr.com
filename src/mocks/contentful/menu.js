import { pagesMap } from './pages'

export const childSubmenu = [
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Resume / Education',
      page: pagesMap['/resume/education'],
      submenu: undefined,
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Resume / Work Experience',
      page: pagesMap['/resume/work-experience'],
      submenu: undefined,
      url: undefined,
    },
  },
]

export const submenu = [
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Home Item',
      page: pagesMap['/'],
      submenu: undefined,
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Blog',
      page: pagesMap['/blog'],
      submenu: undefined,
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Resume',
      submenu: childSubmenu,
      page: pagesMap['/resume'],
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Portfolio',
      page: pagesMap['/portfolio'],
      submenu: undefined,
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO About',
      page: pagesMap['/about'],
      submenu: undefined,
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO External link',
      submenu: undefined,
      page: undefined,
      url: 'http://www.google.com',
    },
  },
  {
    fields: {
      alias: 'Google',
      name: 'Menu - DCR.IO External link with alias',
      submenu: undefined,
      page: undefined,
      url: 'http://www.google.com',
    },
  },
]

export default {
  fields: {
    name: 'Top Menu',
    submenu,
  },
}
