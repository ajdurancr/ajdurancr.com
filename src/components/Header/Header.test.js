import React from 'react'
import { shallow } from 'enzyme'

import Header from './Header'
import menu from '../../mocks/contentful/menu'

describe('Header', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Header pageUrl="/test-url" menu={menu} socialMedia={{ testProp: 'test value' }} />)

    expect(wrapper).toMatchSnapshot()
  })
})
