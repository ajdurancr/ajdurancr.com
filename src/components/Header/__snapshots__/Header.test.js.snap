// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="header header"
>
  <div
    className="container"
  >
    <div
      className="topMenu"
    >
      <Logo
        className="logo"
        onlyIcon={false}
      />
      <Component
        testProp="test value"
      />
    </div>
    <Menu
      menu={
        Object {
          "name": "Top Menu",
          "submenu": Array [
            Object {
              "fields": Object {
                "alias": undefined,
                "name": "Menu - DCR.IO Home Item",
                "page": Object {
                  "fields": Object {
                    "name": "Home",
                    "sections": Array [
                      Object {
                        "fields": Object {
                          "components": Array [
                            Object {
                              "fields": Object {
                                "componentDescription": "This is a test component",
                                "contentType": "Test Component",
                                "name": "TestComponent",
                                "props": Array [
                                  Object {
                                    "fields": Object {
                                      "name": "title",
                                      "value": "This is a test title",
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propText",
                                        },
                                      },
                                      "id": "74gEMnRKSOxejniOwjkHYN",
                                    },
                                  },
                                  Object {
                                    "fields": Object {
                                      "name": "children",
                                      "value": Array [
                                        Object {
                                          "fields": Object {
                                            "componentDescription": "This is a child component",
                                            "contentType": "Test Component",
                                            "name": "ChildComponent",
                                          },
                                          "sys": Object {
                                            "contentType": Object {
                                              "sys": Object {
                                                "id": "component",
                                              },
                                            },
                                            "id": "4gfjlQfm3x9ShZKO8gdYqx",
                                          },
                                        },
                                      ],
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propComponents",
                                        },
                                      },
                                      "id": "3hwiQ7OcvmctEdQPye6exu",
                                    },
                                  },
                                  Object {
                                    "fields": Object {
                                      "name": "markdownField",
                                      "value": "Test value",
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propMarkdown",
                                        },
                                      },
                                      "id": "1RezpPBXf2buGYIccqQw0i",
                                    },
                                  },
                                ],
                              },
                              "sys": Object {
                                "contentType": Object {
                                  "sys": Object {
                                    "id": "component",
                                  },
                                },
                                "id": "6neuJJsr2mbdKli0VjWnCn",
                              },
                            },
                          ],
                          "name": "Main Section",
                        },
                        "sys": Object {
                          "id": "sectionId",
                        },
                      },
                    ],
                    "url": "/",
                  },
                },
                "submenu": undefined,
                "url": undefined,
              },
            },
            Object {
              "fields": Object {
                "alias": undefined,
                "name": "Menu - DCR.IO Blog",
                "page": Object {
                  "fields": Object {
                    "name": "Blog",
                    "sections": Array [
                      Object {
                        "fields": Object {
                          "components": Array [
                            Object {
                              "fields": Object {
                                "componentDescription": "This is a test component",
                                "contentType": "Test Component",
                                "name": "TestComponent",
                                "props": Array [
                                  Object {
                                    "fields": Object {
                                      "name": "title",
                                      "value": "This is a test title",
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propText",
                                        },
                                      },
                                      "id": "74gEMnRKSOxejniOwjkHYN",
                                    },
                                  },
                                  Object {
                                    "fields": Object {
                                      "name": "children",
                                      "value": Array [
                                        Object {
                                          "fields": Object {
                                            "componentDescription": "This is a child component",
                                            "contentType": "Test Component",
                                            "name": "ChildComponent",
                                          },
                                          "sys": Object {
                                            "contentType": Object {
                                              "sys": Object {
                                                "id": "component",
                                              },
                                            },
                                            "id": "4gfjlQfm3x9ShZKO8gdYqx",
                                          },
                                        },
                                      ],
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propComponents",
                                        },
                                      },
                                      "id": "3hwiQ7OcvmctEdQPye6exu",
                                    },
                                  },
                                  Object {
                                    "fields": Object {
                                      "name": "markdownField",
                                      "value": "Test value",
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propMarkdown",
                                        },
                                      },
                                      "id": "1RezpPBXf2buGYIccqQw0i",
                                    },
                                  },
                                ],
                              },
                              "sys": Object {
                                "contentType": Object {
                                  "sys": Object {
                                    "id": "component",
                                  },
                                },
                                "id": "6neuJJsr2mbdKli0VjWnCn",
                              },
                            },
                          ],
                          "name": "Main Section",
                        },
                        "sys": Object {
                          "id": "sectionId",
                        },
                      },
                    ],
                    "url": "/blog",
                  },
                },
                "submenu": undefined,
                "url": undefined,
              },
            },
            Object {
              "fields": Object {
                "alias": undefined,
                "name": "Menu - DCR.IO Resume",
                "page": Object {
                  "fields": Object {
                    "name": "Resume",
                    "sections": undefined,
                    "url": "/resume",
                  },
                },
                "submenu": Array [
                  Object {
                    "fields": Object {
                      "alias": undefined,
                      "name": "Menu - DCR.IO Resume / Education",
                      "page": Object {
                        "fields": Object {
                          "name": "Education",
                          "sections": Array [
                            Object {
                              "fields": Object {
                                "components": Array [
                                  Object {
                                    "fields": Object {
                                      "componentDescription": "This is a test component",
                                      "contentType": "Test Component",
                                      "name": "TestComponent",
                                      "props": Array [
                                        Object {
                                          "fields": Object {
                                            "name": "title",
                                            "value": "This is a test title",
                                          },
                                          "sys": Object {
                                            "contentType": Object {
                                              "sys": Object {
                                                "id": "propText",
                                              },
                                            },
                                            "id": "74gEMnRKSOxejniOwjkHYN",
                                          },
                                        },
                                        Object {
                                          "fields": Object {
                                            "name": "children",
                                            "value": Array [
                                              Object {
                                                "fields": Object {
                                                  "componentDescription": "This is a child component",
                                                  "contentType": "Test Component",
                                                  "name": "ChildComponent",
                                                },
                                                "sys": Object {
                                                  "contentType": Object {
                                                    "sys": Object {
                                                      "id": "component",
                                                    },
                                                  },
                                                  "id": "4gfjlQfm3x9ShZKO8gdYqx",
                                                },
                                              },
                                            ],
                                          },
                                          "sys": Object {
                                            "contentType": Object {
                                              "sys": Object {
                                                "id": "propComponents",
                                              },
                                            },
                                            "id": "3hwiQ7OcvmctEdQPye6exu",
                                          },
                                        },
                                        Object {
                                          "fields": Object {
                                            "name": "markdownField",
                                            "value": "Test value",
                                          },
                                          "sys": Object {
                                            "contentType": Object {
                                              "sys": Object {
                                                "id": "propMarkdown",
                                              },
                                            },
                                            "id": "1RezpPBXf2buGYIccqQw0i",
                                          },
                                        },
                                      ],
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "component",
                                        },
                                      },
                                      "id": "6neuJJsr2mbdKli0VjWnCn",
                                    },
                                  },
                                ],
                                "name": "Main Section",
                              },
                              "sys": Object {
                                "id": "sectionId",
                              },
                            },
                          ],
                          "url": "/resume/education",
                        },
                      },
                      "submenu": undefined,
                      "url": undefined,
                    },
                  },
                  Object {
                    "fields": Object {
                      "alias": undefined,
                      "name": "Menu - DCR.IO Resume / Work Experience",
                      "page": Object {
                        "fields": Object {
                          "name": "Work Experience",
                          "sections": Array [
                            Object {
                              "fields": Object {
                                "components": Array [
                                  Object {
                                    "fields": Object {
                                      "componentDescription": "This is a test component",
                                      "contentType": "Test Component",
                                      "name": "TestComponent",
                                      "props": Array [
                                        Object {
                                          "fields": Object {
                                            "name": "title",
                                            "value": "This is a test title",
                                          },
                                          "sys": Object {
                                            "contentType": Object {
                                              "sys": Object {
                                                "id": "propText",
                                              },
                                            },
                                            "id": "74gEMnRKSOxejniOwjkHYN",
                                          },
                                        },
                                        Object {
                                          "fields": Object {
                                            "name": "children",
                                            "value": Array [
                                              Object {
                                                "fields": Object {
                                                  "componentDescription": "This is a child component",
                                                  "contentType": "Test Component",
                                                  "name": "ChildComponent",
                                                },
                                                "sys": Object {
                                                  "contentType": Object {
                                                    "sys": Object {
                                                      "id": "component",
                                                    },
                                                  },
                                                  "id": "4gfjlQfm3x9ShZKO8gdYqx",
                                                },
                                              },
                                            ],
                                          },
                                          "sys": Object {
                                            "contentType": Object {
                                              "sys": Object {
                                                "id": "propComponents",
                                              },
                                            },
                                            "id": "3hwiQ7OcvmctEdQPye6exu",
                                          },
                                        },
                                        Object {
                                          "fields": Object {
                                            "name": "markdownField",
                                            "value": "Test value",
                                          },
                                          "sys": Object {
                                            "contentType": Object {
                                              "sys": Object {
                                                "id": "propMarkdown",
                                              },
                                            },
                                            "id": "1RezpPBXf2buGYIccqQw0i",
                                          },
                                        },
                                      ],
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "component",
                                        },
                                      },
                                      "id": "6neuJJsr2mbdKli0VjWnCn",
                                    },
                                  },
                                ],
                                "name": "Main Section",
                              },
                              "sys": Object {
                                "id": "sectionId",
                              },
                            },
                          ],
                          "url": "/resume/work-experience",
                        },
                      },
                      "submenu": undefined,
                      "url": undefined,
                    },
                  },
                ],
                "url": undefined,
              },
            },
            Object {
              "fields": Object {
                "alias": undefined,
                "name": "Menu - DCR.IO Portfolio",
                "page": Object {
                  "fields": Object {
                    "name": "Portfolio",
                    "sections": Array [
                      Object {
                        "fields": Object {
                          "components": Array [
                            Object {
                              "fields": Object {
                                "componentDescription": "This is a test component",
                                "contentType": "Test Component",
                                "name": "TestComponent",
                                "props": Array [
                                  Object {
                                    "fields": Object {
                                      "name": "title",
                                      "value": "This is a test title",
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propText",
                                        },
                                      },
                                      "id": "74gEMnRKSOxejniOwjkHYN",
                                    },
                                  },
                                  Object {
                                    "fields": Object {
                                      "name": "children",
                                      "value": Array [
                                        Object {
                                          "fields": Object {
                                            "componentDescription": "This is a child component",
                                            "contentType": "Test Component",
                                            "name": "ChildComponent",
                                          },
                                          "sys": Object {
                                            "contentType": Object {
                                              "sys": Object {
                                                "id": "component",
                                              },
                                            },
                                            "id": "4gfjlQfm3x9ShZKO8gdYqx",
                                          },
                                        },
                                      ],
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propComponents",
                                        },
                                      },
                                      "id": "3hwiQ7OcvmctEdQPye6exu",
                                    },
                                  },
                                  Object {
                                    "fields": Object {
                                      "name": "markdownField",
                                      "value": "Test value",
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propMarkdown",
                                        },
                                      },
                                      "id": "1RezpPBXf2buGYIccqQw0i",
                                    },
                                  },
                                ],
                              },
                              "sys": Object {
                                "contentType": Object {
                                  "sys": Object {
                                    "id": "component",
                                  },
                                },
                                "id": "6neuJJsr2mbdKli0VjWnCn",
                              },
                            },
                          ],
                          "name": "Main Section",
                        },
                        "sys": Object {
                          "id": "sectionId",
                        },
                      },
                    ],
                    "url": "/portfolio",
                  },
                },
                "submenu": undefined,
                "url": undefined,
              },
            },
            Object {
              "fields": Object {
                "alias": undefined,
                "name": "Menu - DCR.IO About",
                "page": Object {
                  "fields": Object {
                    "name": "About",
                    "sections": Array [
                      Object {
                        "fields": Object {
                          "components": Array [
                            Object {
                              "fields": Object {
                                "componentDescription": "This is a test component",
                                "contentType": "Test Component",
                                "name": "TestComponent",
                                "props": Array [
                                  Object {
                                    "fields": Object {
                                      "name": "title",
                                      "value": "This is a test title",
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propText",
                                        },
                                      },
                                      "id": "74gEMnRKSOxejniOwjkHYN",
                                    },
                                  },
                                  Object {
                                    "fields": Object {
                                      "name": "children",
                                      "value": Array [
                                        Object {
                                          "fields": Object {
                                            "componentDescription": "This is a child component",
                                            "contentType": "Test Component",
                                            "name": "ChildComponent",
                                          },
                                          "sys": Object {
                                            "contentType": Object {
                                              "sys": Object {
                                                "id": "component",
                                              },
                                            },
                                            "id": "4gfjlQfm3x9ShZKO8gdYqx",
                                          },
                                        },
                                      ],
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propComponents",
                                        },
                                      },
                                      "id": "3hwiQ7OcvmctEdQPye6exu",
                                    },
                                  },
                                  Object {
                                    "fields": Object {
                                      "name": "markdownField",
                                      "value": "Test value",
                                    },
                                    "sys": Object {
                                      "contentType": Object {
                                        "sys": Object {
                                          "id": "propMarkdown",
                                        },
                                      },
                                      "id": "1RezpPBXf2buGYIccqQw0i",
                                    },
                                  },
                                ],
                              },
                              "sys": Object {
                                "contentType": Object {
                                  "sys": Object {
                                    "id": "component",
                                  },
                                },
                                "id": "6neuJJsr2mbdKli0VjWnCn",
                              },
                            },
                          ],
                          "name": "Main Section",
                        },
                        "sys": Object {
                          "id": "sectionId",
                        },
                      },
                    ],
                    "url": "/about",
                  },
                },
                "submenu": undefined,
                "url": undefined,
              },
            },
            Object {
              "fields": Object {
                "alias": undefined,
                "name": "Menu - DCR.IO External link",
                "page": undefined,
                "submenu": undefined,
                "url": "http://www.google.com",
              },
            },
            Object {
              "fields": Object {
                "alias": "Google",
                "name": "Menu - DCR.IO External link with alias",
                "page": undefined,
                "submenu": undefined,
                "url": "http://www.google.com",
              },
            },
          ],
        }
      }
      pageUrl="/test-url"
    />
  </div>
</div>
`;
